services/explorer-indexer/src/modules/pbft/pbft.service.ts

```diff
@@ -34,14 +34,20 @@ export default class PbftService {
     this.isRedisConnected = state;
   }
 
-  public async getSavedPbftPeriods() {
-    return (
-      await this.pbftRepository.find({
-        select: {
-          number: true,
-        },
-      })
-    )?.map((entry) => entry.number);
+  public async getMissingPbftPeriods() {
```

Comment:
Hi @rattrap . So, I went on and changed the logic a bit so that the calculation is entirely delegated to the database, which is a major improvement. Thanks for the series generation example, it was a really good source of inspiration. I took it one step forward and tweaked it so that it returns all the `pbftPeriods` missing.